src/clusterfuzz/_internal/bot/tasks/trials.py

```diff
@@ -61,17 +68,36 @@ def __init__(self):
       for config in trials_config:
         if config['app_name'] != app_name:
           continue
-        self.trials[config['app_args']] = config['probability']
+        self.trials[config['app_args']] = AppArgs(config['probability'],
+                                                  config.get('contradicts', []))
     except Exception as e:
       logs.log_warn('Unable to parse config file: %s' % str(e))
       return
 
-  def setup_additional_args_for_app(self):
+  def doesnt_asymmetrically_contradicts(self, app_args, existing_contradictions,
+                                        trial_args):
+    return (app_args not in existing_contradictions and
+            (not self.trials[app_args].contradicts or
+             not any(flag in self.trials[app_args].contradicts
+                     for flag in trial_args)))
```

Comment:
I think it'd be more readable to only extract lines 80-82 in their own function and give it a simpler name, e.g. new_arg_contradicts, so that you can write
not new_arg_contradicts(new_args, existing_args)

Another way to make this more readable is inverting the logic in each expression below and use guard clauses:
if (reason one to not select this flag):
 continue
if (reason two to not select this flag):
 continue
...

If you structure it like that, you probably don't need to extract a function. Each if-statement could get a comment that helps understanding.

Reply:
done, I have inverted the logic and moved it to the loop with comments

Reply:
I have removed the `not self.trials[app_args].contradicts` condition since I do not think it is necessary anymore